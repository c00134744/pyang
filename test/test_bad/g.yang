module g {
    namespace "urn:g";
    prefix g;
    
    grouping ww {
        list a {
            leaf b { type int32; }
        }
    }

    container f {
        uses ww; // error: list a needs a key
        leaf bar {
            type keyref {
                path "../h/b"; // error: b not key
            }
        }
        list h {
            key a;
            leaf a {
                type int32;
            }
            leaf b {
                type int32;
            }
        }
    }

    rpc foo {
        input d { // error: bad argument
            container x;
        }
    }
    augment "/foo/d/x" { // error: d/x not found
        leaf a {
            type int32;
        }
    }
}