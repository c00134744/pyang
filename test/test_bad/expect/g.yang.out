g.yang:12: error: NEED_KEY_USES
g.yang:15: error: KEYREF_NOT_LEAF_KEY
g.yang:30: error: UNEXPECTED_ARGUMENT
g.yang:34: error: NODE_NOT_FOUND
g.yang:34: error: NODE_NOT_FOUND
